### ISSUE TITLE
Agent HTTP server limitations

### ISSUE THREAD

**Issue by ryanblenis** [Ylianst/MeshAgent#29]
Hi Brian,

A quick question on the HTTP server within the Mesh Agent.

Easiest way is with an example:

```
http.createServer(function (req, res) {
                try {
                    res.writeHead(200, {'Content-Type': 'text/plain'});
                    res.write('Hello World!');
                    res.end();
                    dbg('ended test');
                } catch(e) { dbg('caught ' + e ); }
            }).listen(8089);
````

In the agent core. Then going to http://127.0.0.1:8089 and looking for "Hello World!", however the process hangs seemingly indefinitely, however if I restart the Mesh Agent, I get the "Hello World!" I was expecting.

I know you've basically made the agent _look_ like it is NodeJS, but given you have the module there and it is in use for sockets, I thought I could do something like this- but maybe just the "res.end()" isn't implemented which is what is causing the issue, so when the process gets killed it finally spits out the "Hello World!" data I was looking for. (Event though my debug line 'ended test' prints successfully, indicating that res.end() does exist).

Thoughts?

Thanks in advance!

**Comment by ryanblenis** [comment]
Actually, it looks like adding a 
```
req.on('end', function () {  });
```
actually produces the response as expected. Thanks!

**Comment by krayon007** [comment]
Out of curiousity, what does your client side code look like? Normally, you hook up the 'response' event of the clientRequest object.

**Comment by ryanblenis** [comment]
Actually that was purely a test to see if it could be done. I didn't go much further than what I had posted. I've just got a few projects in mind that I was trying to see if I could use the mesh agent's local http server to listen for events, such as for facilitating inter-mesh chat functionality via external app (via electron or even powershell).

 I'm currently working on monitoring integration and was testing with Zabbix clients and zabbix active agents routing through the mesh agent, having the mesh agent work as an "offline" zabbix proxy to capture metrics when offline, but I'm still in the research phase because active agents in zabbix can't run remote commands natively- which seems like a stopgap for what I'm trying to accomplish (service/event log monitoring and response) unless I separate that part and have the alert captured, then respond within the mesh agent itself to take action (such as triggering a script to run), and I don't see a great way to make the zabbix agents passive without modifying the zabbix agent, which I was trying to stay away from for long-term viability. 

I'm sure I'll be back to the same type of test with another agent (Nagios, Icinga, etc.) as soon as I find a feature-set that matches what I'm looking for.
